### DISCUSSION TITLE
Documentation for % and the |> tokens?

### DISCUSSION THREAD

**Discussion by olatheander** [KittyCAD/modeling-app#4490] [General]
Hi,

I've played around with the modeler after listening to the episode in Developer Voices. I did a honest try browsing the docs for the KCL language but I can find a clear definition of what's the meaning of the `%` and the `|>` tokens?

E.g. in a KCL like:
```
  |> startProfileAt([0, 0], %)
```
What does it mean to start a line with `|>` and what does the `%` mean?

**Comment by jtran** [comment]
```
a
  |> b(%)
  |> c(%)
```

is the same as

```
c(b(a))
```

The `%` indicates where to substitute the previous value. So

```
  |> startProfileAt([0, 0], %)
```

uses the result from the previous line (probably a plane) as the second argument to startProfileAt.

**Reply by olatheander** [reply]
Hi, thanks for the clarification. However, maybe I just did look at the wrong place but I could not find it documented.

I mostly looked here: https://zoo.dev/docs/kcl
